boost-maven-plugin/src/main/java/io/openliberty/boost/liberty/LibertyPackageMojo.java

```diff
@@ -131,15 +166,25 @@ private void validateSpringBootUberJAR() throws MojoExecutionException {
      * 
      * @throws MojoExecutionException
      */
-    private void copySpringBootUberJar() throws MojoExecutionException {
+    private void copySpringBootUberJar(File springBootUberJar, boolean attach) throws MojoExecutionException {
         try {
-            if (!SpringBootUtil.copySpringBootUberJar(project.getArtifact().getFile(), BoostLogger.getInstance())) {
-                File springJar = new File(
-                        SpringBootUtil.getBoostedSpringBootUberJarPath(project.getArtifact().getFile()));
-                if (net.wasdev.wlp.common.plugins.util.SpringBootUtil.isSpringBootUberJar(springJar)) {
-                    getLog().debug("Copying back Spring Boot Uber JAR as project artifact.");
-                    FileUtils.copyFile(springJar, project.getArtifact().getFile());
+            File springBootUberJarCopy = null;
+            if (springBootUberJar != null) {
```

Comment:
If this test fails (and it's null), it seems to me like we should throw a MojoFailureExc (or ExecutionExc).   If all we're doing is avoiding an NPE.. well, I'd rather get an NPE than unexpected behavior we never planned for.

Reply:
`springBootUberJar` being null doesn't necessarily indicate a failure. This is for the case when a user is running the SB build a second time without cleaning. In that case we would not want to copy, but we would want to copy back (lines 179 to 183) given they are not using the classifier (attach is true).